walk-in-interview/src/main/webapp/applicant-interested-list/script.js

```diff
@@ -0,0 +1,193 @@
+
+  const jobPageData = 
+    await getInterestedJobs(DEFAULT_PAGE_SIZE, DEFAULT_PAGE_INDEX)
+      .catch((error) => {
+        console.error('error fetching job listings', error);
+        setErrorMessage(/* errorMessageElementId= */ 'error-message',
+            /* msg= */ STRINGS['get-jobs-error-message'],
+            /* include default msg= */ false);
+      });
+  
+  displayInterestedJobListings(jobPageData);
+}
+
+/**
+ * Makes GET request to retrieve all the interested job listings
+ * belonging to the current applicant user. 
+ * This function is called when the interest page is loaded.
+ * 
+ * @param {int} pageSize The number of jobs for one page.
+ * @param {int} pageIndex The page index (starting from 0).
+ * @return {Object} The data returned from the servlet.
+ */
+function getInterestedJobs(pageSize, pageIndex) {
+  const params = `pageSize=${pageSize}&pageIndex=${pageIndex}`;
+    
+  fetch(`${API['applicant-interested-list']}?${params}`, {
+    method: 'GET',
+    headers: {'Content-Type': 'application/json'},
+  })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log('data', data);
+        // Resets the error (there might have been an error msg from earlier)
+        setErrorMessage(/* errorMessageElementId= */ 'error-message',
+            /* msg= */ '', /* includesDefaultMsg= */ false);
+        return data;
+      });
+}
+
+/**
+ * This will add all the interested job listings onto the page.
+ * 
+ * @param {Object} jobPageData The details to be shown on the page.
+ */
+function displayInterestedJobListings(jobPageData) {
+  const jobListingsElement = document.getElementById('job-listings');
+  const jobShowing = document.getElementById('job-listings-showing');
+
+  /* reset the list so we don't render the same jobs twice */
+  jobListingsElement.innerHTML = '';
+  jobShowing.innerHTML = '';
+
+  if (jobPageData === undefined ||
+    !jobPageData.hasOwnProperty('jobList') ||
+    jobPageData['jobList'].length === 0) {
+    setErrorMessage(/* errorMessageElementId= */ 'error-message',
+        /* msg= */ STRINGS['no-jobs-error-message'],
+        /* includesDefaultMsg= */ false);
+    return;
+  }
+
+  const jobListings = jobPageData['jobList'];
+
+  jobListings.forEach((job) => {
+    jobListingsElement.appendChild(buildJobElement(job));
+  });
+
+  jobShowing.innerText = `${jobPageData['range'].minimum} -` +
+    ` ${jobPageData['range'].maximum} ${STRINGS['job-listings-showing']} ` +
+    `${jobPageData['totalCount']}`;
+}
+
+/**
+ * Builds the job element given the job details from the servlet response.
+ * 
+ * @param {Object} job The job to be displayed.
+ * @return {Element} The job listing element.
+ */
+function buildJobElement(job) {
+  const jobPostPreviewTemplate =
+    document.getElementById('job-listing-template');
+
+  const jobPostPreview =
+    jobPostPreviewTemplate.cloneNode( /* and child elements */ true);
+  jobPostPreview.setAttribute('id', job['jobId']);
```

Comment:
your id should be guaranteed unique across your html, i recommend you add a prefix like: `job-listing-id-$job['jobId']`